modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationChanger.java

```diff
@@ -736,7 +739,7 @@ private static void dropUnnecessarilyDeletedKeys(Map<String, Serializable> allCh
                 .filter(k -> !localRoots.data.values().containsKey(k))
                 .collect(Collectors.toList());
 
-        unnecessarilyDeletedKeys.forEach(allChanges::remove);
+        allChanges.keySet().removeIf(unnecessarilyDeletedKeys::contains);
```

Comment:
This is not what I meant, can we use `removeIf` without the `unnecessarilyDeletedKeys` keys?

Reply:
Ah, I see